lib/net.js

```diff
@@ -1392,6 +1395,14 @@ Server.prototype.listen = function() {
   var options = normalized[0];
   var cb = normalized[1];
 
+  if (this._serverListenHasBeenCalled === true) {
+    var er = new Error('listen method has been called more than once.');
+    er.code = 'EINVAL';
+    return this.emit('error', er);
+  }
+
+  this._serverListenHasBeenCalled = true;
```

Comment:
What if the `server._handle` has been manually set to a non-listening wrap though?